Code/Framework/AzToolsFramework/Tests/Prefab/Benchmark/Propagation/PropagationBenchmarkFixture.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#if defined(HAVE_BENCHMARK)
+
+#include <AzToolsFramework/Prefab/PrefabDomUtils.h>
+#include <AzToolsFramework/ToolsComponents/EditorInspectorComponent.h>
+#include <Prefab/Benchmark/Propagation/PropagationBenchmarkFixture.h>
+
+namespace Benchmark
+{
+    void PropagationBenchmarkFixture::UpdateTemplate()
+    {
+        PrefabDom updatedPrefabDom;
+        PrefabDomUtils::StoreInstanceInPrefabDom(*m_instanceCreated, updatedPrefabDom);
+        PrefabDom& enclosingTemplatePrefabDom = m_prefabSystemComponent->FindTemplateDom(m_instanceCreated->GetTemplateId());
+        enclosingTemplatePrefabDom.CopyFrom(updatedPrefabDom, enclosingTemplatePrefabDom.GetAllocator());
+        m_instanceUpdateExecutorInterface->AddTemplateInstancesToQueue(m_instanceCreated->GetTemplateId(), *m_instanceCreated);
+    }
+
+    void PropagationBenchmarkFixture::UpdateComponent(benchmark::State& state)
```

Comment:
The shuffle around calling `PauseTiming` and `ResumeTiming` in these seems like it could be abstracted for a bit of extra safety e.g.
```
void PropagationBenchmarkFixture::RunPropagationBenchmark(AZStd::function<void()> runBeforePropagation, AZStd::function<void()> runAfterPropagation)
{
    for ([[maybe_unused]] auto _ : state)
    {
        state.PauseTiming();
        if (runBeforePropagation)
        {
            runBeforePropagation();
        }
        state.ResumeTiming();
        m_instanceUpdateExecutorInterface->UpdateTemplateInstancesInQueue();
        state.PauseTiming();
        if (runAfterPropagation)
        {
            runAfterPropagation();
        }
    }
}
```

Reply:
After doing some auditing, it does look like PauseTiming and ResumeTiming are skewing the results by a lot. When I removed them for one of the runs with a large number of entities, the time measured almost halved. I'd rather let the time be skewed by some setup operations we do instead of functions like PauseTiming and ResumeTiming which we can't control. I'm removing them.